Users enhancement

rodrigoprestesmachado opened the pull request.

lucasTovo reviewed (changes requested):
I suggest changing it from step 3 onwards, dividing the next action into a container to handle the success and failure case and then handle the other cases.

nikolasFreitas reviewed (changes requested):
I think the alternative cases must change. I think it should be re-planned.

I prefer the success cases always first. But there is more things to review in alternative cases.

evandroLutz reviewed (changes requested):
I suggest splitting the alternative cases in a more structured way. Success cases first, with alternatives of success linked with an elses of failure, for each situation.

rodrigoprestesmachado committed:
#25 starting to organize the files

rodrigoprestesmachado committed:
Users enhancement #25

rodrigoprestesmachado committed:
Users enhancement #25

rodrigoprestesmachado committed:
Users enhancement #25

rodrigoprestesmachado committed:
Users enhancement #25

rodrigoprestesmachado committed:
Users enhancement #25

ricardowaldow committed:
fix: #27 Fixing Architecture Issue

- Now delete user endpoint call UseCase layer.

ricardowaldow committed:
fix: Generate Password and Config

- Generate Password method divided in two.
- Unnecessary comments in application properties removed.

rodrigoprestesmachado committed:
Users enhancement #25

rodrigoprestesmachado committed:
Users enhancement #25

ricardowaldow committed:
docs: #26 Documentation updated

- Endpoints that not had docs are now supported by help info.

ricardowaldow committed:
fix: Missing Annotations

- "Produces" and "Consume" annotations added where they was missing.

rodrigoprestesmachado committed:
Users enhancement #25
